coursebuilder/modules/explorer/_static/js/course.js

```diff
@@ -15,18 +15,46 @@ $(document).ready(function documentReady() {
   });
 
   var sidebarToggle = false;
-  $('body').on('click', '#hideCourseSidenav', function () {
+  $("body").on("click", "#hideCourseSidenav", function() {
     sidebarToggle = !sidebarToggle;
-    var sidebarWidth = sidebarToggle ? $('.course-accordion').width() : 0;
+    var sidebarWidth = sidebarToggle ? $(".course-accordion").width() : 0;
 
-    $('.course-accordion').toggleClass('hide-sidenav');
-    $('.course-card').animate({'margin-left': sidebarWidth / 2 + 'px'}, 300);
-    $('.course-accordion').animate({'margin-left': '-' + sidebarWidth + 'px'}, 300);
+    $(".course-accordion").toggleClass("hide-sidenav");
+    $(".course-card").animate({ "margin-left": sidebarWidth / 2 + "px" }, 300);
+    $(".course-accordion").animate(
+      { "margin-left": "-" + sidebarWidth + "px" },
+      300
+    );
   });
 
-  $('.course-accordion .collapsible').clone().appendTo('#courseMobileSidebar');
-  $('.collapsible').collapsible();
-  $('body').on('click', '#closeSideNav', function () {
-    $('.button-collapse').sideNav('hide');
+  $(".course-accordion .collapsible").clone().appendTo("#courseMobileSidebar");
+  $(".collapsible").collapsible();
+  $("body").on("click", "#closeSideNav", function() {
+    $(".button-collapse").sideNav("hide");
+  });
+
+  var parentElementClass = ['item',
+      'owl-carousel owl-theme course-carousel owl-loaded owl-drag',
+      'gcb-border-box', 'owl-item', 'owl-dot active', 'owl-next', 'owl-prev',
+    'owl-prev disabled', 'owl-next disabled', 'owl-dot', 'owl-dots'];
+
+  $(".course-carousel").on("click", function(event) {
+    $(document.documentElement).off("keyup");
+    var owlCarousel = $(event.target).closest(".owl-carousel");
+    var target = $(event.target);
+    $(document.documentElement).on("keyup", function (event) {
+      var carouselContent = target.closest().context.parentElement.className;
```

Comment:
@chegejames Basically I tested many scenarios involving adding google docs, images, code snippet, texts input fields and videos and targeted the classes surrounding an entire carousel. So I am only concerned with classes within the carousel container. the Ids of each content can change but those classes are always there because they are outside the main content of the carousel. Finding the first parent of target tells me which class the user is in. I am using the actual target to pin point the location of its first parent element which is what is needed to stop event bubbling if needed to other elements in the carousel

Reply:
@chegejames To the best of my knowledge, this is the best and most robust way of achieving this.